beacon-chain/state/state-native/field_trie.go

```diff
@@ -15,6 +16,14 @@ var (
 	ErrEmptyFieldTrie   = errors.New("empty field trie")
 )
 
+// SliceAccessor describes an interface for a multivalue slice composite
+// object that returns both the shared multivalue slice along with the
+// particular state instance we are referencing.
+type SliceAccessor[O multi_value_slice.Identifiable] interface {
```

Comment:
Addressed point 1 and 3, not so sure about this one:

```
I am not sure if this is desirable, but if we make the generic type any, then the interface can be used outside of a mvslice context.
```
The reason we have this interface is solely because the multivalue slice is not a native type so we need to use it in some of the paths of the field trie. We can change it to `any` , if we ever come across a new type in the future we might need. But I think being specific to multivalue slice here is fine for now.